src/dodal/devices/undulator_dcm.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+from numpy import argmin, loadtxt, ndarray
+from ophyd import Component, Device, Signal
+from ophyd.status import Status
+
+from dodal.devices.DCM import DCM
+from dodal.devices.undulator import Undulator, UndulatorGapAccess
+from dodal.log import LOGGER
+
+
+class AccessError(Exception):
+    pass
+
+
+def _get_energy_distance_table(lookup_table_path: str) -> ndarray:
+    return loadtxt(lookup_table_path, comments=["#", "Units"])
+
+
+def _get_closest_gap_to_dcm_energy(
+    dcm_energy: float, energy_to_distance_table: ndarray
+) -> float:
```

Comment:
Nit: I would remove reference to the `dcm` here. Just `get_closest_gap_for_energy`